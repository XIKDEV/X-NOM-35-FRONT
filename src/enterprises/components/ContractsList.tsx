import { FC } from 'react';
import { Card, List } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFileArrowDown } from '@fortawesome/free-solid-svg-icons';
import { IContractsListProps } from '../../interfaces';
import {
  bodyContractStyle,
  listContractsStyleDesktop,
  listContractsStyleMobile,
  listItemContractsStyle,
} from '../ui';

export const ContractsList: FC<IContractsListProps> = ({ data, isMobile }) => {
  return (
    <List
      style={isMobile ? listContractsStyleMobile : listContractsStyleDesktop}
      dataSource={data}
      renderItem={(item) => (
        <List.Item style={listItemContractsStyle}>
          <Card bodyStyle={bodyContractStyle} className="enterprise-contract">
            <div className="enteprise-contract-container flex-row-center space-between">
              <p>{item.contract}</p>
              <p>{item.date}</p>
              <button>
                <FontAwesomeIcon icon={faFileArrowDown} />
              </button>
            </div>
          </Card>
        </List.Item>
      )}
    />
  );
};
