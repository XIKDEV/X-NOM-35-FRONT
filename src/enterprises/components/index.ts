export * from './ContractsList';
export * from './EnterprisesCard';
export * from './HeadEnterpriseInfo';
export * from './InfoEnterprise';
