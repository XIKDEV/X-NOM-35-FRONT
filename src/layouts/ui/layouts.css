main {
  display: flex;
}

.sidebar {
  width: 228px;
  height: 100vh;

  background-color: var(--dark);

  justify-content: space-between;
}
.sidebar-logo {
  width: 94px;
  margin-top: 47px;

  cursor: pointer;
}
.sidebar-bttns {
  width: 100%;
}
.sidebar-bttns:last-child {
  margin-bottom: 47px;
}
