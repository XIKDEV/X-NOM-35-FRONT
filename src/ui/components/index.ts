export * from './CardMobile';
export * from './FormItemInput';
export * from './FormItemInputPassword';
export * from './FormItemSelect';
export * from './FormItemSelectCoordinate';
export * from './FormItemUpload';
export * from './FormsButtons';
export * from './FormsDrawer';
export * from './HeaderMobile';
export * from './HeaderPage';
export * from './InfoDrawerButtons';
