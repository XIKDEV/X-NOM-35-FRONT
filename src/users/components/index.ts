export * from './HeaderUserInfo';
export * from './InfoUser';
export * from './UsersCard';
