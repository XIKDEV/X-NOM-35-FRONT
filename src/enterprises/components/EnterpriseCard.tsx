import { FC } from 'react';
import { Card } from 'antd';
import Meta from 'antd/es/card/Meta';
import { ICardProps } from '../../interfaces';
import { enterprisesConstants } from '../../constants';
import { enterprisesCardBodyStyle, enterprisesMetaStyle } from '../ui';

export const EnterpriseCard: FC<ICardProps> = ({
  enterpriseName,
  responsible,
  logoEnterprise,
}) => {
  return (
    <Card
      className="card-enterprise br-8"
      hoverable
      bodyStyle={enterprisesCardBodyStyle}
      cover={
        <img
          alt={`${enterprisesConstants.logoFrom} ${enterpriseName}`}
          src={logoEnterprise}
        />
      }
    >
      <Meta
        title={
          <p className="card-labels" style={enterprisesMetaStyle.title}>
            {enterpriseName}
          </p>
        }
        description={
          <p className="card-labels" style={enterprisesMetaStyle.description}>
            {enterprisesConstants.responsibleEnterprise} {responsible}
          </p>
        }
      />
    </Card>
  );
};
