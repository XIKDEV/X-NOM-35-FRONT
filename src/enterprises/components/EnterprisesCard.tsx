import { FC } from 'react';
import { Card } from 'antd';
import Meta from 'antd/es/card/Meta';
import { ICardProps } from '../../interfaces';
import { cardMetaStyle } from '../../ui/styles';
import { enterprisesCardBodyStyle } from '../ui';

export const EnterprisesCard: FC<ICardProps> = ({
  srcImage,
  title,
  subtitle,
  alt,
  handleOpenDrawerInfo,
}) => {
  return (
    <Card
      className="card-enterprise br-8"
      hoverable
      bodyStyle={enterprisesCardBodyStyle}
      cover={<img src={srcImage} alt={alt} />}
      onClick={handleOpenDrawerInfo}
    >
      <Meta
        title={
          <p className="card-labels" style={cardMetaStyle.title}>
            {title}
          </p>
        }
        description={
          <p className="card-labels" style={cardMetaStyle.description}>
            {subtitle}
          </p>
        }
      />
    </Card>
  );
};
