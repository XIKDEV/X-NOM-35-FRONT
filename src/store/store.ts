import { configureStore } from '@reduxjs/toolkit';

import { authSlice } from './auth';
import { requestSlice } from './request';
import { usersSlice } from './users';

export const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    users: usersSlice.reducer,
    request: requestSlice.reducer,
  },
});
