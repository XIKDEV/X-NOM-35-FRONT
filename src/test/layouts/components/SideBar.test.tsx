import { render } from '@testing-library/react';
// import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { SideBar } from '../../../layouts/components';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';

describe('Test in <SideBar />', () => {
  test('should render correctly when isMobile is true', () => {
    const { queryByTestId } = render(
      <Provider store={store}>
        <Router>
          <SideBar isMobile={true} isCollapse={true} handleSidebar={() => {}} />
        </Router>
      </Provider>
    );
    expect(queryByTestId('sidebar-open-close')).toBeNull();
    expect(queryByTestId('sidebar-logo')).toBeTruthy();
    expect(queryByTestId('sider-navigation')).toBeTruthy();
    expect(queryByTestId('sider-logout')).toBeTruthy();
  });

  // test('should toggle sidebar collapse when button is clicked', () => {
  //   const handleSidebar = jest.fn();
  //   const { getByTestId } = render(
  //     <Router>
  //       <SideBar
  //         isMobile={false}
  //         isCollapse={true}
  //         handleSidebar={handleSidebar}
  //       />
  //     </Router>
  //   );
  //   const button = getByTestId('sidebar-open-close');
  //   fireEvent.click(button);
  //   expect(handleSidebar).toHaveBeenCalledTimes(1);
  // });

  // test('should render the logo image correctly', () => {
  //   const { container } = render(
  //     <Router>
  //       <SideBar isMobile={false} isCollapse={true} handleSidebar={() => {}} />
  //     </Router>
  //   );
  //   const logoImage = container.querySelector('[data-testid="sidebar-logo"]');
  //   expect(logoImage).not.toBeNull();
  //   if (logoImage) {
  //     expect(container.contains(logoImage)).toBe(true);
  //     expect(logoImage.getAttribute('src')).toBe(
  //       '../../../public/XIK_VerdeTransparente.png'
  //     );
  //     expect(logoImage.getAttribute('alt')).toBe('XikDev');
  //   }
  // });

  // test('should render correctly', () => {
  //   const { asFragment } = render(
  //     <Router>
  //       <SideBar isMobile={false} isCollapse={true} handleSidebar={() => {}} />
  //     </Router>
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });

  // test('should toggle style of button based on isCollapse', () => {
  //   const handleSidebarMock = jest.fn();
  //   const { getByTestId, rerender } = render(
  //     <Router>
  //       <SideBar
  //         isMobile={false}
  //         isCollapse={true}
  //         handleSidebar={handleSidebarMock}
  //       />
  //     </Router>
  //   );
  //   const button = getByTestId('sidebar-open-close');
  //   const buttonStyle = getComputedStyle(button);
  //   expect(buttonStyle.right).toBe('');
  //   rerender(
  //     <Router>
  //       <SideBar
  //         isMobile={false}
  //         isCollapse={false}
  //         handleSidebar={handleSidebarMock}
  //       />
  //     </Router>
  //   );
  //   expect(buttonStyle.right).toBe('');
  // });
});
