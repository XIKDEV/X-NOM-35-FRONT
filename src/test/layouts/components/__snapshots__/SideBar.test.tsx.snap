// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SideBar /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sider-content flex-column-center space-between"
  >
    <button
      class="sidebar-open-close"
      data-testid="sidebar-open-close"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-right "
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style="transition: all 0.2s ease-in-out;"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
          fill="currentColor"
        />
      </svg>
    </button>
    <img
      alt="XikDev"
      class="sidebar-logo logo-collapse"
      data-testid="sidebar-logo"
      src="../../../public/XIK_VerdeTransparente.png"
    />
    <section
      class="sider-navigation flex-column-center"
      data-testid="sider-navigation"
    >
      <a
        class="button-sidebar flex-row-center"
        data-testid="button-sidebar"
        href="/nom035/enterprises"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-building "
          data-icon="building"
          data-prefix="far"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style="margin-right: 0px;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 48c-8.8 0-16 7.2-16 16V448c0 8.8 7.2 16 16 16h80V400c0-26.5 21.5-48 48-48s48 21.5 48 48v64h80c8.8 0 16-7.2 16-16V64c0-8.8-7.2-16-16-16H64zM0 64C0 28.7 28.7 0 64 0H320c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm88 40c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16v48c0 8.8-7.2 16-16 16H104c-8.8 0-16-7.2-16-16V104zM232 88h48c8.8 0 16 7.2 16 16v48c0 8.8-7.2 16-16 16H232c-8.8 0-16-7.2-16-16V104c0-8.8 7.2-16 16-16zM88 232c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16v48c0 8.8-7.2 16-16 16H104c-8.8 0-16-7.2-16-16V232zm144-16h48c8.8 0 16 7.2 16 16v48c0 8.8-7.2 16-16 16H232c-8.8 0-16-7.2-16-16V232c0-8.8 7.2-16 16-16z"
            fill="currentColor"
          />
        </svg>
      </a>
      <a
        class="button-sidebar flex-row-center"
        data-testid="button-sidebar"
        href="/nom035/users"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user "
          data-icon="user"
          data-prefix="far"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style="margin-right: 0px;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M304 128a80 80 0 1 0 -160 0 80 80 0 1 0 160 0zM96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM49.3 464H398.7c-8.9-63.3-63.3-112-129-112H178.3c-65.7 0-120.1 48.7-129 112zM0 482.3C0 383.8 79.8 304 178.3 304h91.4C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7H29.7C13.3 512 0 498.7 0 482.3z"
            fill="currentColor"
          />
        </svg>
      </a>
    </section>
    <section
      class="sider-logout flex-column-center"
      data-testid="sider-logout"
    >
      <a
        class="button-sidebar flex-row-center"
        data-testid="button-sidebar"
        href="/auth/login"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-power-off "
          data-icon="power-off"
          data-prefix="fas"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style="margin-right: 0px;"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V256c0 17.7 14.3 32 32 32s32-14.3 32-32V32zM143.5 120.6c13.6-11.3 15.4-31.5 4.1-45.1s-31.5-15.4-45.1-4.1C49.7 115.4 16 181.8 16 256c0 132.5 107.5 240 240 240s240-107.5 240-240c0-74.2-33.8-140.6-86.6-184.6c-13.6-11.3-33.8-9.4-45.1 4.1s-9.4 33.8 4.1 45.1c38.9 32.3 63.5 81 63.5 135.4c0 97.2-78.8 176-176 176s-176-78.8-176-176c0-54.4 24.7-103.1 63.5-135.4z"
            fill="currentColor"
          />
        </svg>
      </a>
    </section>
  </div>
</DocumentFragment>
`;
